src/lib/utils/loadstor.h

```diff
@@ -55,6 +56,8 @@
 
 namespace Botan {
 
+static_assert(std::endian::native == std::endian::big || std::endian::native == std::endian::little);
```

Comment:
Apparently, it is possible that `std::endian::native` is neither `big` nor `little`. At least that's what [the STL example in std::endian](https://en.cppreference.com/w/cpp/types/endian) implies. 😲 Perhaps we should add an error message to this static assert?

```suggestion
static_assert(std::endian::native == std::endian::big || std::endian::native == std::endian::little,
             "mixed-endian systems are not supported");
```

Reply:
Yeah. C++'s continued insistence on supporting systems that haven't been commercially produced in decades is really obnoxious, especially when it comes at the cost of additional complexity for actual software running on actual computers.

Between 523b2a4ca48fa5 and 3dfa8f679bbc `CPUID` would throw upon initialization if it detected a mixed endian system at runtime.

Reply:
> ..., especially when it comes at the cost of additional complexity for actual software running on actual computers.

I guess that would have been okay, if at least the `std::endian` enum had explicitly listed `mixed` so that this is sort-of obvious. But then again: if we wouldn't have considered the mixed case out of ignorance, it would have probably "just worked" (™️). Until an actual mixed system would have come along, no?